Does not compile in Node >= 4

To reproduce:

```
$ git clone https://github.com/GoogleCloudPlatform/gcloud-node-todos.git
$ cd gcloud-node-todos
$ node -v
v6.2.1
$ npm i
...
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/protagonist/src/annotation.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/jdobry/.nvm/versions/node/v6.2.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
gyp ERR! System Darwin 15.5.0
gyp ERR! command "/Users/jdobry/.nvm/versions/node/v6.2.1/bin/node" "/Users/jdobry/.nvm/versions/node/v6.2.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/jdobry/projects/gcloud-node-todos/node_modules/protagonist
gyp ERR! node -v v6.2.1
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
```

The `protagonist` dependency does not compile. @stephenplusplus Do we need newer versions of `todomvc` and `todomvc-api` released with their dependencies upgraded?

`todomvc-api` has a dependency on `dredd` at v0.3.9. The latest stable of `dredd` is v1.0.11. The `protagonist` dependency of `dredd` v0.3.9 does not seem to be compiling.

--- a/package.json
+++ b/package.json
@@ -1,14 +1,14 @@
 {
   "name": "gcloud-node-todos",
-  "description": "todomvc sample for gcloud-node",
+  "description": "TodoMVC sample for gcloud-node",
   "version": "0.2.0",
-  "license": "Apache 2.0",
+  "license": "Apache-2.0",
   "dependencies": {
     "body-parser": "^1.4.3",
     "express": "^4.5.1",
-    "gcloud": "^0.8.1",
+    "gcloud": "^0.36.0",
     "todomvc": "^0.1.0",
-    "todomvc-api": "^0.2.0"
+    "todomvc-api": "^0.2.1"
   },
   "devDependencies": {
     "gulp": "^3.8.8",

--- a/server.js
+++ b/server.js
@@ -38,7 +38,8 @@ api.get('/todos', function(req, res) {
 });
 
 api.get('/todos/:id', function(req, res) {
-  todos.get(req.param('id'), _handleApiResponse(res));
+  var id = parseInt(req.params.id, 10);
+  todos.get(id, _handleApiResponse(res));
 });
 
 api.post('/todos', function(req, res) {
@@ -46,7 +47,8 @@ api.post('/todos', function(req, res) {
 });
 
 api.put('/todos/:id', function(req, res) {
-  todos.update(req.param('id'), req.body, _handleApiResponse(res));
+  var id = parseInt(req.params.id, 10);
+  todos.update(id, req.body, _handleApiResponse(res));
 });
 
 api.delete('/todos', function(req, res) {
@@ -54,7 +56,8 @@ api.delete('/todos', function(req, res) {
 });
 
 api.delete('/todos/:id', function(req, res) {
-  todos.delete(req.param('id'), _handleApiResponse(res, 204));
+  var id = parseInt(req.params.id, 10);
+  todos.delete(id, _handleApiResponse(res, 204));
 });
 
 function _handleApiResponse(res, successStatus) {

--- a/todos.js
+++ b/todos.js
@@ -15,52 +15,71 @@ var gcloud = require('gcloud')({
   credentials: require('./key.json')
 });
 
-var ds = gcloud.datastore.dataset();
+var datastore = gcloud.datastore();
 var LIST_NAME = 'default-list';
 
 function entityToTodo(item) {
   var todo = item.data;
-  todo.id = item.key.path.pop();
+  todo.id = item.key.name || item.key.id;
   return todo;
+}
+
+function saveTodo(key, data, callback) {
+  delete data.id;
+
+  datastore.save({
+    key: key,
+    data: data
+  }, function(err) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    data.id = key.id;
+    callback(null, data);
+  });
 }
 
 module.exports = {
   delete: function(id, callback) {
-    ds.delete(ds.key(['TodoList', LIST_NAME, 'Todo', id]), function(err) {
+    var key = datastore.key(['TodoList', LIST_NAME, 'Todo', id]);
+
+    datastore.delete(key, function(err) {
       callback(err || null);
     });
   },
 
   deleteCompleted: function(callback) {
-    ds.runInTransaction(function(transaction, done) {
-      var q = ds.createQuery('Todo')
-        .hasAncestor(ds.key(['TodoList', LIST_NAME]))
-        .filter('completed =', true);
-      transaction.runQuery(q, function(err, items) {
+    datastore.runInTransaction(function(transaction, done) {
+      var query = transaction.createQuery('Todo')
+        .hasAncestor(datastore.key(['TodoList', LIST_NAME]))
+        .filter('completed', true);
+
+      query.run(function(err, items) {
         if (err) {
           transaction.rollback(done);
           return;
         }
-        var keys = items.map(function(todo) {
+
+        transaction.delete(items.map(function(todo) {
           return todo.key;
-        });
-        transaction.delete(keys, function(err) {
-          if (err) {
-            transaction.rollback(done);
-            return;
-          }
-          done();
-        });
+        }));
+
+        done();
       });
     }, callback);
   },
 
   get: function(id, callback) {
-    ds.get(ds.key(['TodoList', LIST_NAME, 'Todo', id]), function(err, item) {
+    var key = datastore.key(['TodoList', LIST_NAME, 'Todo', id]);
+
+    datastore.get(key, function(err, item) {
       if (err) {
         callback(err);
         return;
       }
+
       if (!item) {
         callback({
           code: 404,
@@ -68,48 +87,36 @@ module.exports = {
         });
         return;
       }
+
       callback(null, entityToTodo(item));
     });
   },
 
   getAll: function(callback) {
-    var q = ds.createQuery('Todo')
-      .hasAncestor(ds.key(['TodoList', LIST_NAME]));
-    ds.runQuery(q, function(err, items) {
+    var query = datastore.createQuery('Todo')
+      .hasAncestor(datastore.key(['TodoList', LIST_NAME]));
+
+    query.run(function(err, items) {
       if (err) {
         callback(err);
         return;
       }
+
       callback(null, items.map(entityToTodo));
     });
   },
 
   insert: function(data, callback) {
+    var key = datastore.key(['TodoList', LIST_NAME, 'Todo']);
+
     data.completed = false;
-    ds.save({
-      key: ds.key(['TodoList', LIST_NAME, 'Todo']),
-      data: data
-    }, function(err, key) {
-      if (err) {
-        callback(err);
-        return;
-      }
-      data.id = key.path.pop();
-      callback(null, data);
-    });
+
+    saveTodo(key, data, callback);
   },
 
   update: function(id, data, callback) {
-    ds.save({
-      key: ds.key(['TodoList', LIST_NAME, 'Todo', id]),
-      data: data
-    }, function(err) {
-      if (err) {
-        callback(err);
-        return;
-      }
-      data.id = id;
-      callback(null, data);
-    });
+    var key = datastore.key(['TodoList', LIST_NAME, 'Todo', id]);
+
+    saveTodo(key, data, callback);
   }
 };

